### ISSUE TITLE
Crew Mgmt - Employee Schedule - modify days off

### ISSUE THREAD

**Issue by JasonSchneider19** [Sanjel-Energy-Services/eService_WorkItems#97]
When i have a worked scheduled for days off, and they would like to (or are willing to) work their days off, i cannot change them to available in the employee schedule:
![image](https://github.com/Sanjel-Energy-Services/eService_WorkItems/assets/95711838/443a916e-4c94-4d8f-8990-d80858619782)

Would like the same functionality to the Off shift days as the available days where i have the following selections available:
![image](https://github.com/Sanjel-Energy-Services/eService_WorkItems/assets/95711838/80391e0c-6038-4468-ba9c-e57559356f6d)

We often have people work their days off and would like the schedule to reflect these details

**Comment by arcurve-cheryl-chan** [comment]
@JasonSchneider19 This seems to be a duplicate of DigitalResourceBoard #1093 - is this how you want to proceed (it was deferred in that work item to Future Phases)?

**Comment by JasonSchneider19** [comment]
@arcurve-cheryl-chan 
Yes this appears to be a duplicate. We are in a future phase so we may as well do it

**Comment by JasonSchneider19** [comment]
@cameliabalan
The first one is all that is required - only need Available
Implement the easiest one

**Comment by JeffFlowerday** [comment]
@JasonSchneider19 do we need to know that they are Off Shift but available.  Or is it OK to just change the status of that day to available?

Are these schedule records used elsewhere in the system or are they informational only?  Just wondering if deleting the Off Shift entry is ok, because that's what makes them available.  No record in the database at all.

FYI @cameliabalan

**Comment by JasonSchneider19** [comment]
@JeffFlowerday 
@awangatsanjel 
So we are starting to use the data in BI for reporting purposes, so if we have an offshift and change him to available, it should be ok to delete the offshift.
What would happen if they wanted to put them to offshift again?

**Comment by JasonSchneider19** [comment]
@JeffFlowerday 
@awangatsanjel 
Adam, need your thoughts on this.
All we really want to do is change the off shift to available so that they can be assigned without warnings. Some people like to work their days off occasionally and need to track it.

**Comment by awangatsanjel** [comment]
@JasonSchneider19 @JeffFlowerday 

- On Shift/Off Shift schedules cannot be deleted, as they are the basis for employee's availiability management and reflect the PayRoll schedules.
- Available is not a schedule type, it is a derived status based on employee arrangement. e.g. If an employee is On Shift today, but not planned, he will be available; if he is assgined to a job, he will be not avaiable; Or he takes a day off, he will not be available. 
- We may use a separate schedule type "On Duty" to mark day(s) as available if an employee works on Off Shift days. We also can use "Off Duty" for not available on On Shift days. 
- In this way, we also can manage resource ad-hoc availability.

Please let me know if it makes sense.

**Comment by JeffFlowerday** [comment]
@awangatsanjel @JasonSchneider19 

So Adam is there anything that needs to be done on your end to handle these new On Duty and Off Duty operation types in the WorkSchedule table?

I assume that only Off Shift has the ability to toggle to "On Duty" and Available(On Shift) has the ability to toggle to "Off Duty"?

Right now I see the following defined in the code, even though 1,2,3 are the only operation_type values in the table, the rest must be derived from other data:

    public enum WorkerScheduleType
    {
        [Display(Name = "Not Applicable")]
        NotApplicable = 0,
        Planned = 1,
        Assigned = 2,

        [Display(Name = "Day Off")]
        DayOff = 3,
        [Display(Name = "On Shift")]
        OnShift = 4,
        [Display(Name = "Off Shift")]
        OffShift = 5,
        Training = 6,
        Vacation = 7
    }

**Comment by cameliabalan** [comment]
Tested the behavior in Test environment #219 and it behaves as described, employees who appear Off Shift can be set to On Duty and back to Off Shift. Also, a message has been added when scheduling crew that includes an employee who appears On Duty to indicate that.
